commit 3b619e22c4601b444ed2d6a5458271f72625ac89
Author: Ard Biesheuvel <ardb@kernel.org>
Date:   Thu Oct 27 17:59:08 2022 +0200

    arm64: implement dynamic shadow call stack for Clang
    
    Implement dynamic shadow call stack support on Clang, by parsing the
    unwind tables at init time to locate all occurrences of PACIASP/AUTIASP
    instructions, and replacing them with the shadow call stack push and pop
    instructions, respectively.
    
    This is useful because the overhead of the shadow call stack is
    difficult to justify on hardware that implements pointer authentication
    (PAC), and given that the PAC instructions are executed as NOPs on
    hardware that doesn't, we can just replace them without breaking
    anything. As PACIASP/AUTIASP are guaranteed to be paired with respect to
    manipulations of the return address, replacing them 1:1 with shadow call
    stack pushes and pops is guaranteed to result in the desired behavior.
    
    Signed-off-by: Ard Biesheuvel <ardb@kernel.org>
    Reviewed-by: Sami Tolvanen <samitolvanen@google.com>
    Tested-by: Sami Tolvanen <samitolvanen@google.com>
    Link: https://lore.kernel.org/r/20221027155908.1940624-4-ardb@kernel.org
    Signed-off-by: Will Deacon <will@kernel.org>

diff --git a/arch/arm64/kernel/irq.c b/arch/arm64/kernel/irq.c
index 38dbd3828f13..9d8eaab742a4 100644
--- a/arch/arm64/kernel/irq.c
+++ b/arch/arm64/kernel/irq.c
@@ -41,7 +41,7 @@ static void init_irq_scs(void)
 {
 	int cpu;
 
-	if (!IS_ENABLED(CONFIG_SHADOW_CALL_STACK))
+	if (!scs_is_enabled())
 		return;
 
 	for_each_possible_cpu(cpu)